src/ui/common/src/components/tables/virtualizedTable.tsx

```diff
@@ -0,0 +1,252 @@
+import { Box, Typography } from '@mui/material';
+import { styled, Theme } from '@mui/material/styles';
+import TableCell from '@mui/material/TableCell';
+import clsx from 'clsx';
+import * as React from 'react';
+import {
+  AutoSizer,
+  Column,
+  Table,
+  TableCellRenderer,
+  TableHeaderProps,
+} from 'react-virtualized';
+
+const classes = {
+  flexContainer: 'ReactVirtualizedDemo-flexContainer',
+  tableRow: 'ReactVirtualizedDemo-tableRow',
+  tableRowHover: 'ReactVirtualizedDemo-tableRowHover',
+  tableCell: 'ReactVirtualizedDemo-tableCell',
+  noClick: 'ReactVirtualizedDemo-noClick',
+};
+
+const columnWidthMultiplier = 13;
+
+const styles = ({ theme }: { theme: Theme }) =>
+  ({
+    // temporary right-to-left patch, waiting for
+    // https://github.com/bvaughn/react-virtualized/issues/454
+    '& .ReactVirtualized__Table__headerRow': {
+      ...(theme.direction === 'rtl' && {
+        paddingLeft: '0 !important',
+      }),
+      ...(theme.direction !== 'rtl' && {
+        paddingRight: undefined,
+      }),
+    },
+    '& .ReactVirtualized__Table__headerColumn': {
+      ...{
+        marginRight: '0px',
+      },
+    },
+    '& .ReactVirtualized__Table__rowColumn': {
+      ...{
+        marginRight: '0px',
+      },
+    },
+    '& .ReactVirtualized__Table__headerColumn:first-of-type': {
+      ...{
+        marginLeft: '0px',
+      },
+    },
+    '& .ReactVirtualized__Table__rowColumn:first-of-type': {
+      ...{
+        marginLeft: '0px',
+      },
+    },
+    [`& .${classes.flexContainer}`]: {
+      display: 'flex',
+      alignItems: 'center',
+      boxSizing: 'border-box',
+    },
+    [`& .${classes.tableRow}`]: {
+      cursor: 'pointer',
+    },
+    [`& .${classes.tableRowHover}`]: {
+      '&:hover': {
+        backgroundColor: theme.palette.grey[200],
+      },
+    },
+    [`& .${classes.tableCell}`]: {
+      flex: 1,
+    },
+    [`& .${classes.noClick}`]: {
+      cursor: 'initial',
+    },
+  } as const);
+
+interface ColumnData {
+  dataKey: string;
+  label: string;
+  type: string;
+  numeric?: boolean;
+  columnWidth?: number;
+}
+
+interface Row {
+  index: number;
+}
+
+interface MuiVirtualizedTableProps {
+  columns: readonly ColumnData[];
+  headerHeight?: number;
+  minColumnWidth?: number;
+  onRowClick?: () => void;
+  rowCount: number;
+  rowGetter: (row: Row) => any;
+  rowHeight?: number;
+}
+
+class MuiVirtualizedTable extends React.PureComponent<MuiVirtualizedTableProps> {
```

Comment:
Is there a reason we're doing a class component here as opposed to a functional component? Also, what is a `PureComponent`? @agiron123, I know you'd like us to use FCs everywhere, so feel free to chime in.

Reply:
I believe this is from a code template we copied over. @agiron123 I'll let you figure out if we should do sth else here. :)

Reply:
@vsreekanti as @cw75 mentioned, this was part of the example code given by MUI. All of the other examples in MUI use functional components. Pure components allow us to have more control over when to re-render a particular component when it's parents update.

See this blog post here for a quick rundown: 
https://dev.to/ashwin_mothilal/difference-between-pure-component-and-functional-component-51mi

If we wish to use a functional component, we can probably do so using the useMemo() hook. I don't really have an opinion either way, we can just leave as is now since things are working quite well. I'd rather use what the example did on MUI, since they probably know more about how MUI and react-virtualized work together than we do :).